proxy/http2/Http2SessionAccept.cc

```diff
@@ -38,9 +39,22 @@ Http2SessionAccept::~Http2SessionAccept()
 void
 Http2SessionAccept::accept(NetVConnection *netvc, MIOBuffer *iobuf, IOBufferReader *reader)
 {
+  AclRecord *session_acl_record = NULL;
+  sockaddr const *client_ip = netvc->get_remote_addr();
+  IpAllow::scoped_config ipallow;
+  if (ipallow && (((session_acl_record = ipallow->match(client_ip)) == NULL) || (session_acl_record->isEmpty()))) {
+    ip_port_text_buffer ipb;
+    Warning("http2 client '%s' prohibited by ip-allow policy", ats_ip_ntop(client_ip, ipb, sizeof(ipb)));
```

Comment:
Actually looking more closely at both cases, both HTTP1 and HTTP2 fail if there is no record.  I'll try to push this test into the super class.